src/util.ts

```diff
@@ -15,6 +15,7 @@ export function range(count: number): number[] {
 
 export const DEFAULT_FRAME_MAX = 1048576
 export const DEFAULT_UNLIMITED_FRAME_MAX = 0
+export const MAX_SHARED_CLIENT_INSTANCES = +(process.env.MAX_SHARED_CLIENT_INSTANCES || 10)
 
```

Comment:
Better use ?? again, and 10 might be a little low. I think that the dotnet uses somewhere around 200, maybe we could do with 100 at least